ui/common/src/commonMain/kotlin/social/androiddev/common/composables/webview/SignInViewModel.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * This file is part of Dodo.
+ *
+ * Dodo is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+ *
+ * Dodo is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Dodo. If not, see <https://www.gnu.org/licenses/>.
+ */
+package social.androiddev.common.composables.webview
+
+import java.net.URI
+
+// TODO USE DI
+// Add tests
+class SignInViewModel(
+    private val server: String
```

Comment:
I think, along side using DI, you could inject the `signInUrl` directly, instead of just `server`. With that, we control the url creation outside of this ViewModel, which will come in handy with testing

Reply:
from my point of view, one of the roles of the SigninViewModel is to get all the necessary elements through use case to build the sign-in URL.

Reply:
We can leave this be for the purpose of this PR, as a follow-up PR I will be connecting this ViewModel to the use case I  created and also setting up DI properly. We can revisit improving this (and adding tests) at that time 😄

Reply:
sounds good 👍